curriculum/challenges/english/01-responsive-web-design/applied-accessibility/improve-chart-accessibility-with-the-figure-element.md

```diff
@@ -34,38 +34,40 @@ Camper Cat is hard at work creating a stacked bar chart showing the amount of ti
 Your code should have one `figure` tag.
 
 ```js
-assert($('figure').length == 1);
+assert.lengthOf(document.querySelectorAll('figure') , 1);
 ```
 
 Your code should have one `figcaption` tag.
 
 ```js
-assert($('figcaption').length == 1);
+assert.lengthOf(document.querySelectorAll('figcaption') , 1);
 ```
 
 Your code should not have any `div` tags.
 
 ```js
-assert($('div').length == 0);
+assert.lengthOf(document.querySelectorAll('div'), 0);
 ```
 
 Your code should not have any `p` tags.
 
 ```js
-assert($('p').length == 0);
+assert.lengthOf(document.querySelectorAll('p') , 0);
 ```
 
 The `figcaption` should be a child of the `figure` tag.
 
 ```js
-assert($('figure').children('figcaption').length == 1);
+const figure = document.querySelector('figure');
+const children = figure.querySelectorAll(`:scope ${'figcaption'}`);
+assert.lengthOf(children, 1);
 ```
 
 Your `figure` element should have a closing tag.
 
 ```js
-assert(
-  code.match(/<\/figure>/g) &&
+assert.isTrue(
+  code.match(/<\/figure>/g) != null &&
     code.match(/<\/figure>/g).length === code.match(/<figure>/g).length
 );
```

Comment:
The more I think about these, the more I believe it'd be better to not use `&&` or `||` in these assertions and split them into two separate assertions.

It's getting confusing very quickly. This one is fine without the `!= null`, due to how `&&` works. If the first operand of `x && y` is _falsy_ it's get returned right away, otherwise the second operand is returned.
```javascript
code.match(/<\/figure>/g) && code.match(/<\/figure>/g).length === code.match(/<figure>/g).length
```
If `code.match(/<\/figure>/g)` doesn't match, it will be `null`, `null` is not `true`, so assertion fails, as expected - good. If it matches, then it's _truthy_, so second operand will be returned `code.match(/<\/figure>/g).length === code.match(/<figure>/g).length` can result as `true`, so all is good. But confusion increases. Which again, could be avoided with two assertions (or maybe just second one with optional chaining, not entirely sure at this moment).

Reply:
I do agree that optional chaining would help clean things up a lot. And maybe it would be better if some assertions were separate.